viewing_party/party.py

```diff
@@ -1,23 +1,179 @@
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating_total = 0
+    num_watched = 0
+    if len(user_data["watched"]) == 0:
+        avg_rating = 0
+    else:
+        for movie_dict in user_data["watched"]:
+            for key, value in movie_dict.items():
+                if key == "rating":
+                    rating_total += value
+                    num_watched += 1
+        avg_rating = rating_total / num_watched
+    return avg_rating
+
+def get_most_watched_genre(user_data):
+    genre_map = {}
+    most_genre = None
+    highest_num = None
+    if len(user_data["watched"]) == 0:
+        return most_genre
+    for movie in user_data["watched"]:
+        for key, value in movie.items():
+            if key == "genre":
+                if value not in genre_map:
+                    genre_map[value] = 1
+                else:
+                    genre_map[value] += 1
+    for genre, num in genre_map.items():
+        if most_genre == None:
+            most_genre = genre
+            highest_num = num
+        elif num > highest_num:
+            most_genre = genre
+            highest_num = num
+        else:
+            continue
 
 
+    return most_genre
+
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    unique_movies = []
+    friend_movies = get_friend_movie_lists(user_data)
+    user_movies = user_data["watched"]
+    for movie_dict in user_movies:
+        if movie_dict not in friend_movies:
+            unique_movies.append(movie_dict)
+
+    return unique_movies
+
+def get_friends_unique_watched(user_data):
+    unique_movies = []
+    friend_movies = get_friend_movie_lists(user_data)
+    user_movies = user_data["watched"]
+    for movie_dict in friend_movies:
+        if movie_dict not in user_movies:
+            unique_movies.append(movie_dict)
+
+    return unique_movies
+
+def get_friend_movie_lists(user_data):
+    friend_movies = []
+    for friend in user_data["friends"]:
+        for movie_dict in friend["watched"]:
+            if movie_dict not in friend_movies:
+                friend_movies.append(movie_dict)
+    return friend_movies
 
-        
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    all_available_recs = get_friends_unique_watched(user_data)
+    user_subscriptions = get_user_subscription(user_data)
+    recs_by_subscription = []
+    for subscription in user_subscriptions:
+        for movie_dict in all_available_recs:
+            if subscription in movie_dict["host"]:
+                recs_by_subscription.append(movie_dict)
+    return recs_by_subscription
+
+def get_user_subscription(user_data):

```

Comment:
Nice job creating this helper function! A slightly more descriptive name would be `get_user_subscriptions` since it can return a list with potentially many items.